test/js/upload-wizard/upload-new-study.test.js

```diff
@@ -1,60 +1,48 @@
 import * as ScpApi from 'lib/scp-api'
 import {
-  EMPTY_STUDY, RAW_COUNTS_FILE, PROCESSED_MATRIX_FILE, METADATA_FILE,
-  CLUSTER_FILE
+  RAW_COUNTS_FILE, PROCESSED_MATRIX_FILE, METADATA_FILE,
+  CLUSTER_FILE, COORDINATE_LABEL_FILE, FASTQ_FILE
 } from './file-info-responses'
-import { UserContext } from 'providers/UserProvider'
-
-/** gets a pointer to the react-select node based on label text
- * This is non-trivial since our labels contain the select,
- * and so a naive getByLabelText will not work.
- * Instead we get the label using getByText and assume
- * the first div inside is the react-select element */
-function getSelectByLabelText(screen, text) {
-  return screen.getByText(text).querySelector('div')
-}
+import { renderWizardWithStudy, getSelectByLabelText, saveButton } from './upload-wizard-test-utils'
+
 
 describe('creation of study files', () => {
-  it('renders the raw counts form and allows updates and save', async () => {
-    const studyInfoSpy = jest.spyOn(ScpApi, 'fetchStudyFileInfo')
-    // pass in a clone of the response since it may get modified by the cache operations
-    studyInfoSpy.mockImplementation(params => {
-      const response = _cloneDeep(EMPTY_STUDY)
-      return Promise.resolve(response)
-    })
+  beforeAll(() => {
+    jest.restoreAllMocks()
+    // This test is long, so allow extra time
+    jest.setTimeout(30000)
+  })
+
+  afterEach(() => {
+    // Restores all mocks back to their original value
+    jest.restoreAllMocks()
+    jest.setTimeout(5000)
+  })
 
+  it('allows upload of all common file types in sequence', async () => {
     const createFileSpy = jest.spyOn(ScpApi, 'createStudyFile')
 
-    const featureFlags = { raw_counts_required_frontend: true }
-    render(<UserContext.Provider value={{ featureFlagsWithDefaults: featureFlags }}>
-      <ReactNotification/>
-      <MockRouter>
-        <RawUploadWizard studyAccession="SCP1" name="Chicken study"/>
-      </MockRouter>
-    </UserContext.Provider>)
-    const saveButton = () => screen.getByTestId('file-save')
-    await waitForElementToBeRemoved(() => screen.getByTestId('upload-wizard-spinner'))
+    await renderWizardWithStudy({ featureFlags: { raw_counts_required_frontend: true } })
     expect(screen.getByText('View study')).toHaveProperty('href', 'http://localhost/single_cell/study/SCP1')
 
-    await testRawCountsUpload({ createFileSpy, saveButton })
-    await testProcessedUpload({ createFileSpy, saveButton })
-    await testMetadataUpload({ createFileSpy, saveButton })
-    await testClusterUpload({ createFileSpy, saveButton })
+    await testRawCountsUpload({ createFileSpy })
+    await testProcessedUpload({ createFileSpy })
+    await testMetadataUpload({ createFileSpy })
+    await testClusterUpload({ createFileSpy })
+    await testSpatialUpload({ createFileSpy })
+    await testCoordinateLabelUpload({ createFileSpy })
+    await testSequenceFileUpload({ createFileSpy })
```

Comment:
Blocker: these shouldn't need to run synchronously, which I suspect is the cause of their excessive slowness.  Could these be wrapped in [`Promise.allSettled()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) or [`Promise.all()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) instead?

Or could these [Jest tests run in parallel or conconcurrently](https://freecontent.manning.com/the-value-of-concurrency-in-tests/)?

Reply:
The whole idea of this test is to test that you can create a study as a user would.  And a user would run them in sequence--and the state of the wizard at each step depends on previous steps.  I agree that we'll want to start parallelizing our jest suites eventually though

Reply:
Makes sense.  Per your note, I've suggested expanding a code comment slightly to ensure it's clear to maintainers.